core/src/main/java/org/jdbi/v3/core/statement/PreparedBatch.java

```diff
@@ -126,8 +108,8 @@ public ResultSetIterable executeAndReturnGeneratedKeys(String... columnNames) {
 
     @Override
     public <R> R execute(ResultProducer<R> producer) {
-        if (parts.isEmpty()) {
-            throw new IllegalStateException("No PreparedBatchParts to execute");
+        if (bindings.isEmpty()) {
```

Comment:
It might be good to add a sanity check to validate that `getBinding()` is empty, to catch users who forget to call `add()`. Mockito does something similar e.g. if you call `verify(mock)` with chaining a method call on the mock.